store/etcd/v2/etcd.go

```diff
@@ -460,16 +462,22 @@ func (s *Etcd) NewLock(key string, options *store.LockOptions) (lock store.Locke
 		}
 	}
 
+	mutexKey := s.normalize(key + "_lock")
+	writeKey := s.normalize(key)
+
 	// Create lock object
 	lock = &etcdLock{
 		client:    s.client,
 		stopRenew: renewCh,
-		mutexKey:  s.normalize(key + "_lock"),
-		writeKey:  s.normalize(key),
+		mutexKey:  mutexKey,
+		writeKey:  writeKey,
 		value:     value,
 		ttl:       ttl,
 	}
 
+	s.mutexKey = mutexKey
+	s.writeKey = writeKey
+
```

Comment:
I think you omitted to update etcd v2 to check the `_lock` entries in the same way as with etcd v3.

Reply:
@abronan I did the modifications for V2 too but the tests were KO with them.
That's why I revert all V2 modifications, the result returned seems to be OK.